### ISSUE TITLE
I updated to Swift 3, and now my menu disappears before completing dismissal animation.

### ISSUE THREAD

**Issue by dilizarov** [Yalantis/Side-Menu.iOS#40]
In this video, I display the bug in action. There really isn't much code to show. It's basically textbook the way you describe to implement this. It worked perfectly before Swift 3.

https://streamable.com/ac6t

### Your Environment

- [x] Version of the component: 2.0
- [x] Swift version: 3
- [x] iOS version: 10
- [x] Device: 6S
- [x] Xcode version: 8
- [x] If you use Cocoapods: 
### Stack

```
   CocoaPods : 1.0.1
        Ruby : ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-darwin14.0]
    RubyGems : 2.2.2
        Host : Mac OS X 10.12 (16A323)
       Xcode : 8.0 (8A218a)
         Git : git version 2.2.1
Ruby lib dir : /Users/dilizarov/.rbenv/versions/2.1.5/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ cee0a231b2467b60bf4de402ed67f0dfce068fbb
               twilio - https://github.com/twilio/cocoapod-specs @ f56585f3f6efb86db536a5123ae196729f05971d
```

### Installation Source

```
Executable Path: /Users/dilizarov/.rbenv/versions/2.1.5/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.0.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '9.0'

source 'https://github.com/CocoaPods/Specs'
source 'https://github.com/twilio/cocoapod-specs'

target 'Chitchat' do
  # Comment this line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!

  pod 'SwiftyJSON', '3.1.0'
  pod 'SCLAlertView-Objective-C', '1.0.3'
  pod 'SDWebImage', '3.8.2'
  pod 'TwilioIPMessagingClient'
  pod 'Alamofire', '4.0.1'
  pod 'SlackTextViewController', '1.9.4'
  pod 'ChameleonFramework/Swift'
  pod 'JGProgressHUD', '1.4'
  pod 'YALSideMenu', '2.0'
  
  # Honestly, I'd prefer to use just one library, but some bugs in ActiveLabel got me to include both of these. In the future, we can refactor some code to just incorporate one. Probably TTTAttributedLabel due to its verbosity
  pod 'ActiveLabel', :git => 'https://github.com/optonaut/ActiveLabel.swift.git', :branch => 'swift-3'
  pod 'TTTAttributedLabel', '2.0.0'
end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['SWIFT_VERSION'] = '3.0'
        end
    end
end
```

- [ ] If you use Carthage: _run `carthage version | pbcopy` and insert here_

**Comment by serejahh** [comment]
Hi @dilizarov. Thanks for reporting the issue! 

Fix is available in `2.0.1`. I missed `self.duration` and local `duration` variable.

**Comment by satish1219** [comment]
please update pod for swift 4
